lucille-core/src/main/java/com/kmwllc/lucille/stage/AddRandomField.java

```diff
@@ -0,0 +1,166 @@
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+
+/**
+ * Adds random data to a document field given parameters
+ * <br>
+ * Config Parameters -
+ * <br>
+ * <p>
+ * <b>inputDataPath</b> (String, Optional) : file path to a text file that stores datapoints to be randomly placed into field,
+ *  defaults to numeric data based on cardinality
+ * </p>
+ * <p>
+ * <b>fieldName</b> (String, Optional) : Field name of field where data is placed, defaults to "data"
+ *  </p>
+ * <p>
+ * <b>cardinality</b> (int, Optional) : size of the subset of datapoints to be grabbed either from
+ *  given datapath or from random numbers
+ * </p>
+ *  <p>
+ * <b>minNumOfTerms</b> (Integer, Optional) : minimum number of terms to be in the field, defaults to null
+ * </p>
+ * <p>
+ * <b>maxNumOfTerms</b> (Integer, Optional) : maximum number of terms to be in the field, defaults to null
+ * </p>
+ * <p>
+ * <b>fieldStructure</b> (FieldType, Optional) : setting for structure of field, default or nested, allows for further settings to be easily added
+ * </p>
+ */
+public class AddRandomField extends Stage {
+
+  enum FieldStructure {
+    DEFAULT, NESTED
+  }
+
+  private final String inputDataPath;
+  private final String fieldName;
+  private Integer cardinality;
+  private Integer minNumOfTerms;
+  private Integer maxNumOfTerms;
+  private final boolean isNested;
+  private List<String> dataArr;
+  private List<String> uniqueValues;
+
+  public AddRandomField(Config config) throws StageException {
+    super(config, new StageSpec().withOptionalProperties("input_data_path", "field_name", "cardinality", "min_num_of_terms",
+        "max_num_of_terms", "is_nested"));
+    this.inputDataPath = ConfigUtils.getOrDefault(config, "input_data_path", null);
+    this.fieldName = ConfigUtils.getOrDefault(config, "field_name", "data");
+    this.minNumOfTerms = ConfigUtils.getOrDefault(config, "min_num_of_terms", null);
+    this.maxNumOfTerms = ConfigUtils.getOrDefault(config, "max_num_of_terms", null);
+    this.isNested = ConfigUtils.getOrDefault(config, "is_nested", false);
+    this.dataArr = null;
+    this.cardinality = null;
+    this.uniqueValues = null;
+
+    if (this.minNumOfTerms == null ^ this.maxNumOfTerms == null) {
+      throw new StageException("Both minimum and maximum number of terms must be specified if either is specified");
+    }
+    if (this.minNumOfTerms == null && this.maxNumOfTerms == null) {
+      this.minNumOfTerms = 1;
+      this.maxNumOfTerms = 1;
+    }
+    if (this.minNumOfTerms > this.maxNumOfTerms) {
+      throw new StageException("Minimum number of terms must be less than or equal to maximum");
+    }
+  }
+
+  @Override
+  public void start() throws StageException {
+    this.dataArr = this.inputDataPath != null ? getFileData(this.inputDataPath) : null;
+    this.cardinality = ConfigUtils.getOrDefault(config, "cardinality",
```

Comment:
wondering if rangeSize would be clearer than "cardinality"

Reply:
Changed name to rangeSize (range_size for config)